include/generic_datatype.h

```diff
@@ -1,16 +1,77 @@
 
-#pragma once
+//#pragma once
 
-#include <functional>
+//#include <functional>
+
+#include <unordered_map>
+#include <vector>
 
 #include "datatype.h"
 #include "exceptions.h"
+#include "generic_sort.h"
 #include "smt_defs.h"
-#include "sort.h"
 
 // using namespace smt;
 namespace smt {
 
+struct selectorComponents
+{
+  std::string name;
```

Comment:
I would expect us to need to also store the corresponding constructor (every selector is associated with a constructor).